src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,107 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_NUMBER = 1.5;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        this.array = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
+    private void growIfArrayFull() {
+        if (size >= array.length) {
+            int newCapacity = (int) (array.length * GROWTH_NUMBER);
+            if (newCapacity <= size) {
+                newCapacity = size + 1;
+            }
+            array = Arrays.copyOf(array, newCapacity);
+        }
+    }
```

Comment:
(Optional!) Not sure if mentors would like it, its not necessary, but i would change that :/

Reply:
What do you mean? You mean - instead of Arrays.copyOf() to use  "loop" to create a new array and copy elements, am I right? or something else?